### DISCUSSION TITLE
how to use Google auth with Expo App

### DISCUSSION THREAD

**Discussion by Bunnavoit** [pocketbase/pocketbase#3885] [Q&A] [answered]
hi , am new here on pocketbase can you tell me is pocketbase goolge auth is support for react native expo app or not ? when i create credential in google developer console it only give me client id and for pocketbase google oauth it require both client secret and client id ? so how can i obtain both Client ID and Client Secret ?
+ inside credential section  choose application type : android

**Comment by ganigeorgiev** [comment, marked as answer]
When creating the Google OAuth2 app, for application type select "Web application", not Android:

![google_app](https://i.imgur.com/OLfhLDd.png)

If you use the ["all-in-one" flow](https://pocketbase.io/docs/authentication/#oauth2-integration) for the `Authorized redirect URIs` field you can use `http://127.0.0.1:8090/api/oauth2-redirect` (if you are testing with ngrok, replace the localhost address with the public generated one from ngrok).

Also note that the "all-in-one" flow requires a realtime connection (SSE), meaning that you'll need to load a `EventSource` polyfill:
```js
// see more in https://github.com/pocketbase/js-sdk#nodejs-via-npm
// for React Native: npm install react-native-sse --save
import eventsource from "react-native-sse";

global.EventSource = eventsource;

...
```

If you have issue with your Expo app in debug mode on Android, then most likely it is caused by `facebook/flipper` plugin. Make sure to check the resources linked in https://github.com/pocketbase/pocketbase/discussions/2484#discussioncomment-6118523.

**Reply by Bunnavoit** [reply]
thank for your explanation really appreciate that

**Reply by Bunnavoit** [reply]
can you give me an example code how to use it in react native expo ? for Client id and Client secret i already put in pocketbase dashboard and google auth is enable now

**Reply by ganigeorgiev** [reply]
For an example you can check the expo link from this older discussion https://github.com/pocketbase/pocketbase/discussions/2958#discussioncomment-6501454.

**Reply by Bunnavoit** [reply]
Thank you
